inc/transfer.class.php

```diff
@@ -474,21 +474,25 @@ function simulateTransfer($items) {
       $this->item_recurs['Software']
             = $this->createSearchConditionUsingArray($this->noneedtobe_transfer['Software']);
 
-      // Move license of software
-      // TODO : should we transfer "affected license" ?
-      $query = "SELECT `id`, `softwareversions_id_buy`, `softwareversions_id_use`
-                FROM `glpi_softwarelicenses`
-                WHERE `softwares_id` IN ".$this->item_search['Software'];
-
-      foreach ($DB->request($query) AS $lic) {
-         $this->addToBeTransfer('SoftwareLicense', $lic['id']);
-
-         // Force version transfer (remove from item_recurs)
-         if ($lic['softwareversions_id_buy'] > 0) {
-            $this->addToBeTransfer('SoftwareVersion', $lic['softwareversions_id_buy']);
-         }
-         if ($lic['softwareversions_id_use'] > 0) {
-            $this->addToBeTransfer('SoftwareVersion', $lic['softwareversions_id_use']);
+      if (count($this->needtobe_transfer['Software'])) {
+         // Move license of software
+         // TODO : should we transfer "affected license" ?
+         $iterator = $DB->request([
+            'SELECT' => ['id', 'softwareversions_id_buy', 'softwareversions_id_use'],
+            'FROM'   => 'glpi_softwarelicenses',
+            'WHERE'  => ['softwares_id' => $this->needtobe_transfer['Software']]
```

Comment:
Query was on  `$this->item_search['Software']`. If updated, count test should be updated as well.

Reply:
My bad. Usage of `$this->needtobe_transfer['Software']` is OK here.